### PR TITLE
Pass creds and output directory as options

### PR DESCRIPTION
**Pull request by philroche**
All github creds can now be passed as options or environment variables.

--config-skeleton no longer requires --config to be set
--output-directory set the directory for the review.html to be saved to.

You can still use environment variables to specify github creds.

### CODE REVIEW

File: src/review_gator/review_gator.py [hunk 1]
```diff
@@ -258,14 +259,17 @@ def get_repo_data(repos):
     return repo_data
 
 
-def render(repos):
+def render(repos, output_directory=None):
```
Comment (fginther) [inline src/review_gator/review_gator.py]: `output_directory` does not to be a kwarg. In the one location it is called, `output_directory` is passed (its default is set via the `click.option` default.

File: src/review_gator/review_gator.py [hunk 2]
```diff
@@ -258,14 +259,17 @@ def get_repo_data(repos):
...
+    abs_templates_path = os.path.join(os.path.dirname(
+            os.path.realpath(__file__)), "templates")
     env = Environment(loader=FileSystemLoader(abs_templates_path))
     tmpl = env.get_template('reviews.html')
-    output_directory = os.environ.get('SNAP_USER_COMMON', "/tmp/review_gator/")
-    output_directory = os.path.realpath(os.path.join(output_directory, "dist"))
+    if not output_directory:
+        output_directory = os.environ.get('SNAP_USER_COMMON',
+                                          "/tmp/review_gator/")
+
```
Comment (fginther) [inline src/review_gator/review_gator.py]: This `if not output_directory` block is not necessary. `output_directory` will always have a default value set via the `click.option` configuration.

File: src/review_gator/clicklib.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from click import Option, UsageError
+
+
+class NotRequiredIf(Option):
```
Comment (fginther) [inline src/review_gator/clicklib.py]: This appears to come from https://stackoverflow.com/questions/44247099/click-command-line-interfaces-make-options-required-if-other-optional-option-is/44349292. Do you mind including a reference here in the code to attribute the source?

File: src/review_gator/review_gator.py [hunk 4]
```diff
@@ -444,15 +452,42 @@ def get_source_info(source):
...
+                   "variable.", default=None)
+@click.option('--github-password', envvar='GITHUB_PASSWORD', required=False,
+              help="Your github password."
+                   "You can also set GITHUB_PASSWORD as an environment "
+                   "variable.", default=None)
+@click.option('--github-token', envvar='GITHUB_TOKEN', required=False,
+              help="Your github api token. If you provide this then you do not"
+                   "need to provide username and password. "
+                   "You can also set GITHUB_TOKEN as an environment "
+                   "variable.", default=None)
```
Comment (fginther) [inline src/review_gator/review_gator.py]: The option / environment variable handling looks much better when done via these `click.options`. Thanks for adding this.

File: src/review_gator/review_gator.py (hunk 1 above)
Comment (philroche) [reply src/review_gator/review_gator.py]: Good catch

File: src/review_gator/review_gator.py (hunk 2 above)
Comment (philroche) [reply src/review_gator/review_gator.py]: Agreed

**Comment by philroche** [comment]
@fginther Thank you. Ready for review again now.

**Pull request merged**